### ISSUE TITLE
False positive in ARM Linter

### ISSUE THREAD

**Issue by spoelly** [github/super-linter#1670]
## Describe the bug
With the ARM linter an error is raised on "serviceEndpoints" that it is a unreferenced variable. I don't understand why an i think it is a false positive:

ARM Template: 
```json
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "existingVirtualNetworkName": {
      "type": "string",      
      "metadata": {
        "description": "Name of the the virtual network in which the subnet is created"
      }
    },
    "existingNSGName": {
      "type": "string",      
      "defaultValue": "",
      "metadata": {
        "description": "Name of the network security group that is going to be attached to the subnet"
      }
    },
    "newSubnetAddressSpace": {
      "type": "string",
      "metadata": {
        "description": "Address space of the new subnet"
      }
    },
    "newSubnetApplicationName": {
      "type": "string",
      "metadata": {
        "description": "Name of application for the new subnet"
      }
    },
    "serviceEndpoints": {
      "type": "string",      
      "defaultValue": "",
      "metadata": {
        "description": "List of service endpoints which needs to be attached to the subnet (use , as separator)"
      }
    }
  },
  "variables": {
    "subnetName": "[concat(parameters('existingVirtualNetworkName'), '/snet-',toLower(parameters('newSubnetApplicationName')))]",
    "NSG": {
      "id": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('existingNSGName'))]"
    },
    "serviceEndpointsArray": "[split(parameters('serviceEndpoints'),',')]",
    "subnetPropertiesEndpoints": {
      "addressPrefix": "[parameters('newSubnetAddressSpace')]",
      "networkSecurityGroup": "[if(not(empty(parameters('existingNSGName'))), variables('NSG'), json('null'))]",
      "privateEndpointNetworkPolicies": "Enabled",
      "privateLinkServiceNetworkPolicies": "Enabled",
      "copy": [
        {
          "name": "serviceEndpoints",
          "count": "[length(variables('serviceEndpointsArray'))]",
          "input": {
            "service": "[if(greater(length(parameters('serviceEndpoints')), 0), variables('serviceEndpointsArray')[copyIndex('serviceEndpoints')], '')]"
          }
        }
      ]
    },
    "subnetPropertiesNoEndpoints": {
      "addressPrefix": "[parameters('newSubnetAddressSpace')]",
      "networkSecurityGroup": "[if(not(empty(parameters('existingNSGName'))), variables('NSG'), json('null'))]",
      "privateEndpointNetworkPolicies": "Enabled",
      "privateLinkServiceNetworkPolicies": "Enabled"
    }
  },
  "resources": [
    {
      "type": "Microsoft.Network/virtualNetworks/subnets",
      "apiVersion": "2019-09-01",
      "name": "[variables('subnetName')]",
      "properties": "[if(greater(length(parameters('serviceEndpoints')), 0), variables('subnetPropertiesEndpoints'), variables('subnetPropertiesNoEndpoints'))]"
    }
  ]
}
```

## Expected behavior
I think this code should pass the linter checks, because of the copyIndex() i think a false positive is raised.

## Steps to Reproduce
docker run \
  -e RUN_LOCAL=true \
  -e ACTIONS_RUNNER_DEBUG=true \
  -e DISABLE_ERRORS=false \
  -e ERROR_ON_MISSING_EXEC_BIT=true \
  -e LINTER_RULES_PATH=. \
  -e MULTI_STATUS=false \
  -e VALIDATE_ALL_CODEBASE=true \
  -v $(pwd):/tmp/lint \
  ghcr.io/github/super-linter:latest

1. Just run it through the linter.

## Logs
2021-06-18 13:35:05 [INFO]   File:[/tmp/lint/Catalog/Subnet/templates/azuredeploy.json]
2021-06-18 13:35:09 [ERROR]   Found errors in [arm-ttk] linter!
2021-06-18 13:35:09 [ERROR]   Error code: 1. Command output:
------

Validating templateszuredeploy.json
  deploymentTemplate
    [+] adminUsername Should Not Be A Literal (98 ms)
    [+] apiVersions Should Be Recent (65 ms)
    [+] artifacts parameter (15 ms)
    [+] Deployment Resources Must Not Be Debug (9 ms)
    [+] Dynamic Variable References Should Not Use Concat (17 ms)
    [+] IDs Should Be Derived From ResourceIDs (18 ms)
    [+] Location Should Not Be Hardcoded (56 ms)
    [+] ManagedIdentityExtension must not be used (6 ms)
    [+] Min And Max Value Are Numbers (9 ms)
    [+] Outputs Must Not Contain Secrets (8 ms)
    [+] Parameters Must Be Referenced (151 ms)
    [+] providers apiVersions Is Not Permitted (3 ms)
    [+] ResourceIds should not contain (15 ms)
    [+] Resources Should Have Location (8 ms)
    [+] Secure String Parameters Cannot Have Default (4 ms)
    [+] Template Should Not Contain Blanks (14 ms)
    [-] Variables Must Be Referenced (39 ms)
        Unreferenced variable: serviceEndpoints

    [+] Virtual Machines Should Not Be Preview (41 ms)
    [+] VM Images Should Use Latest Version (5 ms)
    [+] VM Size Should Be A Parameter (7 ms)

------

**Comment by admiralAwkbar** [comment]
@spoelly 
Hey thanks for the info!

I see your running the super-linter version `3.13.5` which had a much older version of `arm-ttk` installed.

Can you try to run again and use `ghcr.io/github/super-linter:v4`

**Comment by spoelly** [comment]
@admiralAwkbar sorry my bad i am running the latest version had not changed it in the bug template ;)

**Comment by spoelly** [comment]
Why is this closed?
